pkg/sqlfs/hive_writer.go

```diff
@@ -17,13 +17,78 @@ import (
 	"database/sql"
 	"encoding/base64"
 	"fmt"
+	"io"
 	"io/ioutil"
 	"os"
 	"os/exec"
+	"path"
 
 	pb "sqlflow.org/sqlflow/pkg/proto"
 )
 
+func flushToCSV() (func([]byte) error, *os.File, error) {
+	csv, e := ioutil.TempFile("", "sqlflow-sqlfs")
+	if e != nil {
+		return nil, nil, fmt.Errorf("cannot create CSV file: %v", e)
+	}
+
+	row := 0
+	return func(buf []byte) error {
+		if len(buf) > 0 {
+			block := base64.StdEncoding.EncodeToString(buf)
+			_, e := csv.Write([]byte(fmt.Sprintf("%d\001%s\n", row, block)))
+			if e != nil {
+				return fmt.Errorf("cannot flush to CSV, %v", e)
+			}
+			row++
+		}
+		return nil
+	}, csv, nil
+}
+
+func uploadCSVFile(csv *os.File, db *sql.DB, hivePath, table, user, passwd string) func() error {
+	return func() error {
+		defer func() {
+			csv.Close()
+			os.Remove(csv.Name())
+		}()
+
+		hdfsPath := path.Join(hivePath, "sqlfs", table)
+
+		hdfsEnv := os.Environ()
+		if user != "" {
+			hdfsEnv = append(hdfsEnv,
+				fmt.Sprintf("HADOOP_USER_NAME=%s", user),
+				fmt.Sprintf("HADOOP_USER_PASSWORD=%s", passwd))
+		}
+
+		cmd := exec.Command("hdfs", "dfs", "-mkdir", "-p", hdfsPath)
+		cmd.Env = hdfsEnv
+		if _, e := cmd.CombinedOutput(); e != nil {
+			return fmt.Errorf("failed %s: %v", cmd, e)
+		}
+
+		cmd = exec.Command("hdfs", "dfs", "-copyFromLocal", csv.Name(), hdfsPath)
+		cmd.Env = hdfsEnv
+		if _, e := cmd.CombinedOutput(); e != nil {
+			return fmt.Errorf("failed %s: %v", cmd, e)
+		}
+
+		_, e := db.Exec(fmt.Sprintf("LOAD DATA INPATH '%s' OVERWRITE INTO TABLE %s", hdfsPath, table))
+		return e
+	}
+}
+
+func newHiveWriter(db *sql.DB, hivePath, table, user, passwd string) (io.WriteCloser, error) {
+	flush, csv, e := flushToCSV()
```

Comment:
Need to drop and create a table here:
https://github.com/sql-machine-learning/sqlflow/blob/5bf232b52f02f8bfc72862044dc699a51ad6a94c/pkg/sqlfs/table.go#L24-L32